src/overlays/actors/ovl_En_Horse_Normal/z_en_horse_normal.c

```diff
@@ -21,45 +48,635 @@ const ActorInit En_Horse_Normal_InitVars = {
+    s32 pad;
+
+    switch (D_80A6D510[this->animationIdx]) {
+        case 0:
+            func_80A6BD7C(this);
+            this->actor.speedXZ = 0.0f;
+            this->unk_218 = 0.0f;
+            break;
+        case 1:
+            if (Math_Rand_ZeroOne() < 0.1f) {
+                this->unk_218 = 2.0f * Math_Rand_ZeroOne() - 1.0f;
+            }
+            this->actor.speedXZ += this->unk_218;
+            if (this->actor.speedXZ <= 0.0f) {
+                this->actor.speedXZ = 0.0f;
+                this->unk_218 = 0.0f;
+                phi_t0 = 0;
+            } else if (this->actor.speedXZ < 3.0f) {
+                func_80A6B250(this);
+                phi_t0 = 4;
+            } else if (this->actor.speedXZ < 6.0f) {
+                phi_t0 = 5;
+            } else if (this->actor.speedXZ < 8.0f) {
+                phi_t0 = 6;
+            } else {
+                this->actor.speedXZ = 8.0f;
+                phi_t0 = 6;
+            }
+            if (Math_Rand_ZeroOne() < 0.1f ||
+                (this->unk_21E == 0 && (this->actor.bgCheckFlags & 8 || this->bodyCollider.base.maskA & 2 ||
+                                        this->headCollider.base.maskA & 2))) {
+                this->unk_21E += (Math_Rand_ZeroOne() * 30.0f) - 15.0f;
+                if (this->unk_21E > 50) {
+                    this->unk_21E = 50;
+                } else if (this->unk_21E < -50) {
+                    this->unk_21E = -50;
+                }
+            }
+            this->unk_21C += this->unk_21E;
+            if (this->unk_21C < -300) {
+                this->unk_21C = -300;
+            } else if (this->unk_21C > 300) {
+                this->unk_21C = 300;
+            } else if (Math_Rand_ZeroOne() < 0.25f && fabsf(this->unk_21C) < 100.0f) {
+                this->unk_21C = 0;
+                this->unk_21E = 0;
+            }
+            this->actor.posRot.rot.y += this->unk_21C;
+            this->actor.shape.rot.y = this->actor.posRot.rot.y;
+            break;
+        case 2:
+        case 3:
+            break;
+    }
+
+    if (phi_t0 != this->animationIdx || SkelAnime_FrameUpdateMatrix(&this->skin.skelAnime)) {
+        if (phi_t0 != this->animationIdx) {
+            this->animationIdx = phi_t0;
+            this->unk_1E4 &= ~1;
+            this->unk_1E4 &= ~2;
+            if (phi_t0 == 1) {
+                Audio_PlaySoundGeneral(NA_SE_EV_HORSE_GROAN, &this->unk_204, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+            } else if (phi_t0 == 3) {
+                Audio_PlaySoundGeneral(NA_SE_EV_HORSE_NEIGH, &this->unk_204, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+            } else {
+                func_80A6BCEC(this);
+            }
+            SkelAnime_ChangeAnim(&this->skin.skelAnime, sAnimations[this->animationIdx], func_80A6B30C(this), 0.0f,
+                                 SkelAnime_GetFrameCount(&sAnimations[this->animationIdx]->genericHeader), 2, -3.0f);
+            return;
+        }
```

Comment:
wait for someone else's input on this but unfortunately I would put an else here instead of an early return 😅

Reply:
I agree, the blocks look equivalent to me so it should probably be an else